toto/models/matchrule.py

```diff
@@ -0,0 +1,102 @@
+"""
+<Module Name>
+  toto/models/validators.py
+
+<Author>
+  Santiago Torres-Arias <santiago@nyu.edu>
+
+<Started>
+  Nov 18, 2016
+
+<Copyright>
+  See LICENSE for licensing information.
+
+<Purpose>
+  This module provides functions to ensure the syntax of the matchrules is
+  correct.
+
+"""
+from toto.ssl_commons.exceptions import FormatError
+import toto.ssl_crypto.formats
+
+def _validate_match_rule(keywords):
+  """ private helper to verify the syntax of the MATCH matchrule """
+  MATERIAL_OR_PRODUCT = {'PRODUCT', 'MATERIAL'}
+
+  if not isinstance(keywords, list):
+    raise FormatError("this matching rule is not a list")
+
+  if len(keywords) == 5:
+    rule, artifact, target, from_keyword, step = keywords
+    as_keyword = "AS"
+    new_target = target
+
+  elif len(keywords) == 7:
+    rule, artifact, target, from_keyword, step, as_keyword, new_target = keywords
+  else:
+    raise FormatError("Wrong rule format, should be: MATCH (MATERIAL/PRODUCT)"
+      "<target> FROM <step> [AS <new_target>].\n\t"
+      "Got: {}".format(" ".join(keywords)))
+
+  if rule != "MATCH":
+    raise FormatError("Wrong rule to verify! {}".format(rule))
+
+  if from_keyword != "FROM":
+    raise FormatError("FROM should come before step")
+
+  if as_keyword != "AS":
+    raise FormatError("AS should come after the step name")
+
+  if artifact not in MATERIAL_OR_PRODUCT:
+    raise FormatError("Target should be either MATERIAL or PRODUCT!")
+
+def _validate_generic_rule(keywords):
+  """ private helper that verifies the syntax of the other rules """
+
+  VALID_OPERATIONS = {'CREATE', 'MODIFY', 'DELETE',}
+
+  if not isinstance(keywords, list):
+    raise FormatError("this matching rule is not a list")
+
+  if len(keywords) != 2:
+    raise FormatError("Wrong rule format")
+
+  rule, artifact = keywords
+
+  toto.ssl_crypto.formats.PATH_SCHEMA.check_match(artifact)
```

Comment:
Actually this can be a *path pattern*. Since [`PATH_SCHEMA`](https://github.com/secure-systems-lab/ssl_crypto/blob/master/formats.py#L123) only requires it to be any string, we are fine here.